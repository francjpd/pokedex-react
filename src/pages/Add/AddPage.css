.pokedex-pokemon-add {
  display: grid;
  grid-template-rows: 1fr;

  & .title {
    margin: 0;
    text-align: right;
    width: 100%;
    color: var(--tertiary-color);
    font-style: initial;
    font-weight: 100;
    font-size: 20px;
  }

  & .pokedex-pokemon-add__card {
    border: 2px solid var(--tertiary-color);
    border-radius: 6px;

    & .pokedex-pokemon-add__header {
      display: flex;
      align-items: center;
      justify-content: space-between;
      background-color: var(--tertiary-color);
      padding: 8px;

      & .pokedex-pokemon-add__name {
        font-style: italic;
        font-weight: bold;
        text-transform: uppercase;
        font-family: fantasy;
        color: white;
        font-size: 24px;
      }
    }

    & .pokedex-pokemon-add__content {
      display: grid;

      & form {
        display: flex;
        flex-wrap: wrap;
        flex-direction: row;
        justify-content: space-evenly;
        align-items: center;
        & input,
        & select {
          padding: 8px;
          box-sizing: border-box;
          margin: 8px 0;
          border: 1px solid var(--secondary-color);
          border-radius: 4px;
        }
        & .pokedex-pokemon-add__type {
          display: flex;
          flex-direction: column;
        }
        & .pokedex-pokemon-add__submit {
          display: flex;
          justify-content: center;
          width: 100%;
          &,
          input,
          & select {
            max-width: 150px;
            display: block;
          }
        }
      }
    }
  }
}
