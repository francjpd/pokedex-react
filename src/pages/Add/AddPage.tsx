import { useForm } from 'react-hook-form';
import { t } from "i18next";
import './AddPage.css';
import './AddPage.sm.css';
import './AddPage.md.css';
import './AddPage.lg.css';
import './AddPage.xl.css';
import { setPokemon } from '../../features/common/local-storage';

export default function AddPage() {
    const { register, handleSubmit, formState: { errors } } = useForm();
    const onSubmit = (data: any) => {
        setPokemon(data.name, data);
    };

    return (
        <div className="pokedex-pokemon-add">
            <div className="pokedex-pokemon-add__card">
                <div className="pokedex-pokemon-add__header">
                    <div className="pokedex-pokemon-add__name">
                        <>
                            {t('Add a pokemon')}
                        </>
                    </div>
                </div>
                <div className="pokedex-pokemon-add__content">
                    <form onSubmit={handleSubmit(onSubmit)}>
                        <input type="text" arial-label="name" placeholder="name" {...register("name", { required: true })} />
                        <input type="number" arial-label="height" placeholder="height" {...register("height", { required: true, max: 9999, min: 0 })} />
                        <input type="number" arial-label="weight" placeholder="weight" {...register("weight", { required: true, max: 9999, min: 0 })} />
                        <input type="number" arial-label="baseExperience" placeholder="baseExperience" {...register("baseExperience", { required: true, max: 9999, min: 0 })} />
                        <input type="number" arial-label="hp" placeholder="hp" {...register("hp", { required: true, max: 9999, min: 0 })} />
                        <input type="number" arial-label="attack" placeholder="attack" {...register("attack", { required: true, max: 9999, min: 0 })} />
                        <input type="number" arial-label="defense" placeholder="defense" {...register("defense", { required: true, max: 9999, min: 0 })} />
                        <input type="number" aria-label="special-attack" placeholder="special-attack" {...register("special-attack", { required: true, max: 9999, min: 0 })} />
                        <input type="number" aria-label="special-defense" placeholder="special-defense" {...register("special-defense", { required: true, max: 9999, min: 0 })} />
                        <input type="number" arial-label="speed" placeholder="speed" {...register("speed", { required: true, max: 9999, min: 0 })} />
                        <div className='pokedex-pokemon-add__type'>
                            <label>Choose types</label>
                            <select {...register("types", { required: true })} multiple>
                                <option value="bug">bug</option>
                                <option value="dark">dark</option>
                                <option value="dragon">dragon</option>
                                <option value="electric">electric</option>
                                <option value="fire">fire</option>
                                <option value="fairy">fairy</option>
                                <option value="fighting">fighting</option>
                                <option value="flying">flying</option>
                                <option value="ghost">ghost</option>
                                <option value="grass">grass</option>
                                <option value="ground">ground</option>
                                <option value="ice">ice</option>
                                <option value="normal">normal</option>
                                <option value="poison">poison</option>
                                <option value="psychic">psychic</option>
                                <option value="rock">rock</option>
                                <option value="steel">steel</option>
                                <option value="water">water</option>
                            </select>
                        </div>
                        <div className='pokedex-pokemon-add__submit'><input type="submit" aria-label="submit" /></div>
                    </form>
                </div>
            </div>
        </div >
    )
}