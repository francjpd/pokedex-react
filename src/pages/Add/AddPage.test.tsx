import { render } from '@testing-library/react';
import AddPage from './AddPage';
jest.mock('react-router-dom')

describe('AddPage view test', () => {
    let component: any;
    beforeEach(() => {
        component = render(<AddPage />);
    })

    it('should render the AddPage layout correctly', () => {
        expect(component).toMatchSnapshot();

    });

    // it('should validate all fields', () => {
    //     // to do
    // });

})
