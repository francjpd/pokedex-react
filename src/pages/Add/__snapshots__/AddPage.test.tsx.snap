// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddPage view test should render the AddPage layout correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="pokedex-pokemon-add"
      >
        <div
          class="pokedex-pokemon-add__card"
        >
          <div
            class="pokedex-pokemon-add__header"
          >
            <div
              class="pokedex-pokemon-add__name"
            />
          </div>
          <div
            class="pokedex-pokemon-add__content"
          >
            <form>
              <input
                arial-label="name"
                name="name"
                placeholder="name"
                type="text"
              />
              <input
                arial-label="height"
                name="height"
                placeholder="height"
                type="number"
              />
              <input
                arial-label="weight"
                name="weight"
                placeholder="weight"
                type="number"
              />
              <input
                arial-label="baseExperience"
                name="baseExperience"
                placeholder="baseExperience"
                type="number"
              />
              <input
                arial-label="hp"
                name="hp"
                placeholder="hp"
                type="number"
              />
              <input
                arial-label="attack"
                name="attack"
                placeholder="attack"
                type="number"
              />
              <input
                arial-label="defense"
                name="defense"
                placeholder="defense"
                type="number"
              />
              <input
                aria-label="special-attack"
                name="special-attack"
                placeholder="special-attack"
                type="number"
              />
              <input
                aria-label="special-defense"
                name="special-defense"
                placeholder="special-defense"
                type="number"
              />
              <input
                arial-label="speed"
                name="speed"
                placeholder="speed"
                type="number"
              />
              <div
                class="pokedex-pokemon-add__type"
              >
                <label>
                  Choose types
                </label>
                <select
                  multiple=""
                  name="types"
                >
                  <option
                    value="bug"
                  >
                    bug
                  </option>
                  <option
                    value="dark"
                  >
                    dark
                  </option>
                  <option
                    value="dragon"
                  >
                    dragon
                  </option>
                  <option
                    value="electric"
                  >
                    electric
                  </option>
                  <option
                    value="fire"
                  >
                    fire
                  </option>
                  <option
                    value="fairy"
                  >
                    fairy
                  </option>
                  <option
                    value="fighting"
                  >
                    fighting
                  </option>
                  <option
                    value="flying"
                  >
                    flying
                  </option>
                  <option
                    value="ghost"
                  >
                    ghost
                  </option>
                  <option
                    value="grass"
                  >
                    grass
                  </option>
                  <option
                    value="ground"
                  >
                    ground
                  </option>
                  <option
                    value="ice"
                  >
                    ice
                  </option>
                  <option
                    value="normal"
                  >
                    normal
                  </option>
                  <option
                    value="poison"
                  >
                    poison
                  </option>
                  <option
                    value="psychic"
                  >
                    psychic
                  </option>
                  <option
                    value="rock"
                  >
                    rock
                  </option>
                  <option
                    value="steel"
                  >
                    steel
                  </option>
                  <option
                    value="water"
                  >
                    water
                  </option>
                </select>
              </div>
              <div
                class="pokedex-pokemon-add__submit"
              >
                <input
                  aria-label="submit"
                  type="submit"
                />
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="pokedex-pokemon-add"
    >
      <div
        class="pokedex-pokemon-add__card"
      >
        <div
          class="pokedex-pokemon-add__header"
        >
          <div
            class="pokedex-pokemon-add__name"
          />
        </div>
        <div
          class="pokedex-pokemon-add__content"
        >
          <form>
            <input
              arial-label="name"
              name="name"
              placeholder="name"
              type="text"
            />
            <input
              arial-label="height"
              name="height"
              placeholder="height"
              type="number"
            />
            <input
              arial-label="weight"
              name="weight"
              placeholder="weight"
              type="number"
            />
            <input
              arial-label="baseExperience"
              name="baseExperience"
              placeholder="baseExperience"
              type="number"
            />
            <input
              arial-label="hp"
              name="hp"
              placeholder="hp"
              type="number"
            />
            <input
              arial-label="attack"
              name="attack"
              placeholder="attack"
              type="number"
            />
            <input
              arial-label="defense"
              name="defense"
              placeholder="defense"
              type="number"
            />
            <input
              aria-label="special-attack"
              name="special-attack"
              placeholder="special-attack"
              type="number"
            />
            <input
              aria-label="special-defense"
              name="special-defense"
              placeholder="special-defense"
              type="number"
            />
            <input
              arial-label="speed"
              name="speed"
              placeholder="speed"
              type="number"
            />
            <div
              class="pokedex-pokemon-add__type"
            >
              <label>
                Choose types
              </label>
              <select
                multiple=""
                name="types"
              >
                <option
                  value="bug"
                >
                  bug
                </option>
                <option
                  value="dark"
                >
                  dark
                </option>
                <option
                  value="dragon"
                >
                  dragon
                </option>
                <option
                  value="electric"
                >
                  electric
                </option>
                <option
                  value="fire"
                >
                  fire
                </option>
                <option
                  value="fairy"
                >
                  fairy
                </option>
                <option
                  value="fighting"
                >
                  fighting
                </option>
                <option
                  value="flying"
                >
                  flying
                </option>
                <option
                  value="ghost"
                >
                  ghost
                </option>
                <option
                  value="grass"
                >
                  grass
                </option>
                <option
                  value="ground"
                >
                  ground
                </option>
                <option
                  value="ice"
                >
                  ice
                </option>
                <option
                  value="normal"
                >
                  normal
                </option>
                <option
                  value="poison"
                >
                  poison
                </option>
                <option
                  value="psychic"
                >
                  psychic
                </option>
                <option
                  value="rock"
                >
                  rock
                </option>
                <option
                  value="steel"
                >
                  steel
                </option>
                <option
                  value="water"
                >
                  water
                </option>
              </select>
            </div>
            <div
              class="pokedex-pokemon-add__submit"
            >
              <input
                aria-label="submit"
                type="submit"
              />
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
